packages/components/src/search/index.js

```diff
@@ -63,6 +63,8 @@ export class Search extends Component {
 				return usernames;
 			case 'variations':
 				return variations;
+			case 'custom':
+				return this.props.autocompleter;
```

Comment:
Instead of a default prop for `autocompleter`, maybe this would be a good spot to throw an error with a message explaining that an `autocompleter` object is required. Or a custom propType described in https://reactjs.org/docs/typechecking-with-proptypes.html perhaps?

Reply:
Great idea! I've removed the default type in https://github.com/woocommerce/woocommerce-admin/pull/4518/commits/8df16ff1bb77f1ea9b814bbfc29793b4238b92a0 and added the error in https://github.com/woocommerce/woocommerce-admin/pull/4518/commits/5a3b934b1d9fd4b4ec53bb3cd36ca7f8c259dbb9.

Do you feel like we should check the individual properties of the `autocompleter` and throw the same error if some of them are missing?

Also, should that error message be translated? I've found some examples of dev related errors that aren't, so I didn't wrap it in `__`.

Reply:
>Do you feel like we should check the individual properties of the autocompleter and throw the same error if some of them are missing?

I don't think so. Maybe if you went the custom proptype route.

>should that error message be translated?

Good question. I don't think so because a developer will search the codebase for the string.